src/components/cc-order-summary/cc-order-summary.js

```diff
@@ -0,0 +1,233 @@
+import { css, html, LitElement } from 'lit';
+import { dispatchCustomEvent } from '../../lib/events.js';
+import { i18n } from '../../translations/translation.js';
+import '../cc-badge/cc-badge.js';
+import '../cc-button/cc-button.js';
+import '../cc-img/cc-img.js';
+
+/**
+ * @typedef {import('./cc-order-summary.types.js').ConfigurationItem} ConfigurationItem
+ * @typedef {import('./cc-order-summary.types.js').OrderSummaryState} OrderSummaryState
+ */
+
+/**
+ * Displays a summary of a product being currently ordered.
+ *
+ * The UI is composed of:
+ * - global information about the product,
+ * - a configuration in the form of a list of label/value,
+ * - a button to trigger the creation,
+ * - a list of details for additional information.
+ *
+ * @cssdisplay block
+ *
+ * @fires {CustomEvent} cc-order-summary:create - Fires when the create button is clicked.
+ *
+ * @cssprop {FontSize} --cc-order-summary-detail-font-size - The font-size for the list of details (defaults: `0.825em`).
+ * @cssprop {FontWeight} --cc-order-summary-font-weight - Sets the value of the font weight CSS property (defaults: `600`).
+ *
+ * @slot detail - a single piece of information displayed under the main part of the UI. You can insert multiple detail items.
+ */
+
+export class CcOrderSummary extends LitElement {
+  static get properties() {
+    return {
+      orderSummaryState: { type: Object, attribute: 'order-summary-state' },
+    };
+  }
+
+  constructor() {
+    super();
+
+    /** @type {OrderSummaryState} component main state */
+    this.orderSummaryState = null;
+  }
+
+  _onCreateClick() {
+    dispatchCustomEvent(this, 'create');
+  }
+
+  render() {
+    if (this.orderSummaryState == null) {
+      return '';
+    }
+
+    return html`
+      <div class="title">${i18n('cc-order-summary.title')}</div>
+      <div class="summary">${this._renderSummary()}</div>
+      <div class="details-container">
+        <slot name="detail"></slot>
```

Comment:
note: I was about to say "Do we need this slot? Why can't we just put stuff below?" but I guess it's a good way to encapsulate the styles in the component.